code/obj/item/clothing/hats.dm

```diff
@@ -332,6 +333,7 @@ proc/filter_trait_hats(var/type)
 /obj/item/clothing/head/det_hat/gadget
 	name = "DetGadget hat"
 	desc = "Detective's special hat you can outfit with various items for easy retrieval!"
+	item_function_flags = IMMUNE_TO_ACID
```

Comment:
```suggestion
```

Acid immunity is typically only granted to antagonist gear, with almost no exceptions.